# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/GamepadPlatformService.h
# Commit: 5729967eff6d
# Full Hash: 5729967eff6d48c5aa73641f5e89caaf59df9cb7
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-10-09 21:45:45
# Regressor Bug: 1657404
# File Overlap Count: 1
# Description:
#   Bug 1657404 - Use WeakPtr for observer pattern in GamepadPlatformService r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93085
# ==============================================================================

diff -r 694d23b789b1 -r 5729967eff6d dom/gamepad/GamepadPlatformService.h
--- a/dom/gamepad/GamepadPlatformService.h	Fri Oct 09 17:42:38 2020 +0000
+++ b/dom/gamepad/GamepadPlatformService.h	Fri Oct 09 16:42:53 2020 +0000
@@ -12,6 +12,7 @@
 #include <map>
 #include "mozilla/Mutex.h"
 #include "mozilla/StaticPtr.h"
+#include "mozilla/WeakPtr.h"
 
 namespace mozilla {
 namespace dom {
@@ -39,7 +40,7 @@
     MonitoringState() = default;
     ~MonitoringState();
 
-    void AddObserver(RefPtr<GamepadTestChannelParent> aParent);
+    void AddObserver(WeakPtr<GamepadTestChannelParent> aParent);
     void RemoveObserver(GamepadTestChannelParent* aParent);
 
     bool IsMonitoring() const;
@@ -53,7 +54,7 @@
     void Set(bool aIsMonitoring);
 
     bool mIsMonitoring{false};
-    nsTArray<RefPtr<GamepadTestChannelParent>> mObservers;
+    nsTArray<WeakPtr<GamepadTestChannelParent>> mObservers;
 
     friend class GamepadPlatformService;
   };